pkg/config/config.go

```diff
@@ -251,6 +259,9 @@ func (c *Config) UnmarshalYAML(unmarshal func(interface{}) error) error {
 		if rc.WontFixResolution == "" && c.Defaults.WontFixResolution != "" {
 			rc.WontFixResolution = c.Defaults.WontFixResolution
 		}
+		if rc.AutoResolveState == "" && c.Defaults.AutoResolveState != "" {
+			rc.AutoResolveState = c.Defaults.AutoResolveState
+		}
```

Comment:
why cannot you define `AutoResolve` at a receiver level?

Reply:
We can define it at the receiver level. The type is boolean so by default it will be false if we explicitly don't define it. 

This is similar to how `add_group_labels` behaves. The logic of default and overrides seems a little bit counter intuitive when it comes to boolean types. For e.g if we define `auto_resolve: true` at default section, the receiver section would still need to explicitly define `auto_resolve: true`. 

I think what would make more sense in terms of easing this aspect would be having configuration similar to this.
```
defaults:
  auto_resolve:
    state: 'Done'
    
receivers:
  - name: jira-ab
    auto_resolve:
      state: 'Resolved'
  - name: jira-xy
    // no explicit override, so assume defaults. 
    
  ...  
``` 
So in the receiver section logic we can do
```
if c.Defaults.auto_resolve != nil {
  if c.Defaults.auto_resolve.state == "" {
  return fmt.Errorf("bad config")
  }
}
...

for _, rc := range c.Receivers {
  ...
    if rc.auto_resolve != nil {
      if rc.auto_resolve.state == "" {
        return fmt.Errorf("bad config")
      }
    } 
  if rc.auto_resolve == nil && c.Defaults.auto_resolve != nil {
     rc.auto_resolve = c.Defaults.auto_resolve
  }
}

WDYT?

Reply:
Implemented this in [ae42519](https://github.com/bhushanthakur93/jiralert/pull/1/commits/ae42519d1a3171ee49abcfc8a0599c81a8b062af)

Reply:
makes sense, I like it better.